### DISCUSSION TITLE
Will SpinalHDL add support for Scala3 ? (SpinalHDL是否会加入对Scala3的支持?)

### DISCUSSION THREAD

**Discussion by SDTEFU** [SpinalHDL/SpinalHDL#1087] [Q&A]
Will SpinalHDL add support for Scala3 ? (SpinalHDL是否会加入对Scala3的支持?)

**Comment by Dolu1990** [comment]
Hard to tell.
Got the SpinalHDL core to work on scala3 via : 
https://github.com/SpinalHDL/SpinalHDL/tree/scala3-dev

The issue is that scala 3 kinda messed up quite a few things related to internal domain specific language, so the SpinalHDL lib has quite a few issues.
Mostly the Scala compiler crashing (no erroring) XD

So, overall, scala 2 is a better fit to Idsl than scala 3 (at least in some ways)

Currently, scala 3 port is frozen
